ansible/roles/vm_set/files/mux_simulator.py

```diff
@@ -384,16 +403,169 @@ def all_mux_status(vm_set):
             all_mux_status[bridge] = get_mux_status(vm_set, port_index)
         return jsonify(all_mux_status)
     except Exception as e:
-        return jsonify({'err_msg': 'Get all mux status failed, vm_set: {}, exception: {}'.format(vm_set, repr(e))}), 500
+        err_msg = 'Get all mux status failed, vm_set: {}, exception: {}'.format(vm_set, repr(e))
+        app.logger.error('{} {} {}'.format(request.method, request.url, err_msg))
+        return jsonify({'err_msg': err_msg}), 500
+
+
+def _validate_out_ports(data):
+    """Validate the posted data for updating flow action.
+
+    Args:
+        data (dict): Posted json data. Expected:
+                {"out_ports": [<port>, <port>, ...]}
+            where <port> could be "nic", "tor_a" or "tor_b".
+
+    Returns:
+        tuple: Return the result in a tuple. The first item is either True or False. The second item is extra message.
+    """
+    supported_out_ports = ['nic', 'tor_a', 'tor_b']
+    try:
+        assert 'out_ports' in data, 'Missing "out_ports" field'
+        for port in data['out_ports']:
+            assert port in supported_out_ports, 'Unsupported port: "{}", supported: {}'.format(port, supported_out_ports)
+        return True, ''
+    except Exception as e:
+        return False, 'Validate out_ports {} failed with exception: {}'.format(json.dumps(data), repr(e))
+
+
+def update_flow_action_nic(mux_status, action):
+    """Update the action for the flow to "nic".
+
+    Args:
+        mux_status (dict): Current mux status.
+        action (string): The action to be applied to flow. Either "output" or "drop".
+
+    Returns:
+        dict: The new mux status.
+    """
+    in_port = mux_status['active_port']
+    out_port = mux_status['ports']['nic'] if action == 'output' else None
+    action_desc = '{}:"{}"'.format(action, out_port) if out_port else action
+    cmdline = 'ovs-ofctl --name mod-flows {} \'in_port="{}" actions={}\''.format(
+        mux_status['bridge'],
+        in_port,
+        action_desc)
+    run_cmd(cmdline)
+    flow = {in_port: [{'action': action, 'out_port': out_port}]}
+    mux_status['flows'].update(flow)
+    return mux_status
+
+
+def update_flow_action_tor(mux_status, action, tor_ports):
```

Comment:
It's better to change the method name to ```update_flow_action_to_tor``` for better readability.

Reply:
Sure, updated.